### PR TITLE
Fix error on missing author

### PR DESCRIPTION
**Pull request by novascreen**
[This workflow](https://github.com/ecobee/protos/actions/runs/4409547042/jobs/7726023478) ran into a [commit without an author](https://github.com/ecobee/protos/commit/64619f290221299d49e98a61004cd85b80e8aa18) and failed.

I've added a check to make sure the `commit.author` object is defined.

I also updated/switched to `@vercel/ncc` because it wouldn't run on an M1

### CODE REVIEW

File: __tests__/mockResponses.ts [hunk 1]
```diff
@@ -35,10 +35,6 @@ export const successfulComparisonResponse = {
         'https://github.com/octocat/Hello-World/commit/f0c556fd81d38b23ff2b9c7c9d192ff426ff35bbc4',
       commit: {
         message: 'Title of my commit\nBody of my message'
-      },
```
Comment (novascreen) [inline __tests__/mockResponses.ts:38]: the duplicate case is already tested with `octocat` so i removed this one to test the no author case

File: package.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
     "@types/node": "12.7.5",
     "@typescript-eslint/eslint-plugin": "2.3.0",
     "@typescript-eslint/parser": "2.3.0",
-    "@zeit/ncc": "^0.20.5",
+    "@vercel/ncc": "^0.36.1",
```
Comment (psean-eco) [inline package.json:39]: any breaking changes between versions we need to watch out for ?

**Review by psean-eco — APPROVED** [review]
LGTM

File: package.json (hunk 2 above)
Comment (novascreen) [reply package.json:39]: no breaking changes mentioned in the releases

**Pull request merged**